src/framework/circuit.hpp

```diff
@@ -234,8 +200,287 @@ Circuit::Circuit(const inputdata_t &circ, const json_t &qobj_config) : Circuit()
+          first_measure_pos = rpos;
+          has_measure = true;
+        } else if (op.type == OpType::initialize && last_initialize_pos == 0) {
+          last_initialize_pos = rpos;
+        }
+        if (last_ancestor_pos == 0) {
+          last_ancestor_pos = rpos;
+        }
+      } else if (truncation && !ops_to_remove){
+        ops_to_remove = true;
+      }
+    }
+  }
+
+  // Set qubit size and check for truncaiton
+  remapped_qubits = false;
+  if (truncation) {
+    // Generate mapping of original qubits to ancestor set
+    uint_t idx = 0;
+    for (const auto& qubit: qubitset_) {
+      if (!remapped_qubits && idx != qubit) {
+        // qubits will be remapped
+        remapped_qubits = true;
+      }
+      qubitmap_[qubit] = idx;
+      idx++;
+    }
+  }
+
+  // Set qubit and memory size
+  num_memory = (memoryset_.empty()) ? 0 : 1 + *memoryset_.rbegin();
+  num_registers = (registerset_.empty()) ? 0 : 1 + *registerset_.rbegin();
+  if (remapped_qubits) {
+    num_qubits = qubitset_.size();
+  } else {
+    num_qubits = (qubitset_.empty()) ? 0 : 1 + *qubitset_.rbegin();
+  }
+
+  // Check if can sample initialize
+  if (last_initialize_pos > 0 &&
+      ops[last_initialize_pos].qubits.size() < num_qubits) {
+    can_sample_initialize = false;
+    can_sample = false;
+  }
+
+  // Check measurement opt and split tail meas and non-meas ops
+  std::vector<uint_t> tail_pos;
+  std::vector<Op> tail_meas_ops;
+  if (has_measure && can_sample) {
+    std::unordered_set<uint_t> meas_qubits;
+    std::unordered_set<uint_t> modified_qubits;
+
+    for (uint_t pos = first_measure_pos; pos <= last_ancestor_pos; ++pos) {
+      if (ops_to_remove && !ancestor[pos]) {
+        // Skip if not ancestor
+        continue;
+      }
+
+      const auto& op = ops[pos];
+      if (op.conditional) {
+        can_sample = false;
+        break;
+      }
+
+      switch (op.type) {
+        case OpType::measure:
+        case OpType::roerror: {
+          meas_qubits.insert(op.qubits.begin(), op.qubits.end());
+          tail_meas_ops.push_back(op);
+          break;  
+        }
+        case OpType::snapshot:
+        case OpType::save_state:
+        case OpType::save_expval:
+        case OpType::save_expval_var:
+        case OpType::save_statevec:
+        case OpType::save_statevec_dict:
+        case OpType::save_densmat:
+        case OpType::save_probs:
+        case OpType::save_probs_ket:
+        case OpType::save_amps:
+        case OpType::save_amps_sq:
+        case OpType::save_stabilizer:
+        case OpType::save_unitary:
+        case OpType::save_mps:
+        case OpType::save_superop: {
+          can_sample = false;
+          break;
+        }
+        default: {
+          for (const auto &qubit : op.qubits) {
+            if (meas_qubits.find(qubit) != meas_qubits.end()) {
+              can_sample = false;
+              break;
+            }
+          }
+          tail_pos.push_back(pos);
+        }
```

Comment:
```suggestion
          tail_pos.push_back(pos);
          break;
        }
```
?  Not that it matters as it stands, but missing `break` makes me nervous!  (Lots of things make me nervous in C/C++ so perhaps it's just lack of familiarity.)

Reply:
the last default clause of a switch statement doesn't need a break